### PR TITLE
syslog-ng/main: resolve path templates in --help output

### LINKED ISSUE #4138: Not all variable is expanded in the help screen
# syslog-ng
## Version of syslog-ng
```
$ syslog-ng --version
syslog-ng 3 (3.38.1)
Config version: 3.35
Installer-Version: 3.38.1
Revision: 3.38.1-1
Compile-Date: Sep 10 2022 12:22:21
Module-Directory: /usr/lib/syslog-ng/3.38
Module-Path: /usr/lib/syslog-ng/3.38
Include-Path: /usr/share/syslog-ng/include
Available-Modules: rate-limit-filter,pacctformat,confgen,json-plugin,tags-parser,linux-kmsg-format,afprog,pseudofile,basicfuncs,sdjournal,timestamp,dbparser,system-source,tfgetent,disk-buffer,cryptofuncs,syslogformat,csvparser,hook-commands,kvformat,appmodel,afsocket,map-value-pairs,afuser,regexp-parser,affile,cef
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Systemd: on

```

## Platform
Debian Unstable

## Debug bundle
N/A

# Issue
## Failure
$ syslog-ng --help
Usage:
  syslog-ng [OPTION?] syslog-ng

Help Options:
  -h, --help                                                 Show help options
  --help-all                                                 Show all help options
  --help-process                                             Process options
  --help-log                                                 Log options

Application Options:
  -V, --version                                              Display version number (syslog-ng 3 (3.38.1))
  --module-path=<path>                                       Set the list of colon separated directories to search for modules, default=**${exec_prefix}**/lib/syslog-ng/3.38
  --module-registry                                          Display module information
  --no-module-discovery                                      Disable module auto-discovery, all modules need to be loaded explicitly by the configuration
  -S, --seed                                                 Does nothing, the need to seed the random generator is autodetected
  -f, --cfgfile=<config>                                     Set config file name, default=/etc/syslog-ng/syslog-ng.conf
  -R, --persist-file=<fname>                                 Set the name of the persistent configuration file, default=/var/lib/syslog-ng/syslog-ng.persist
  --preprocess-into=output                                   Write the preprocessed configuration file to the file specified and quit
  -s, --syntax-only                                          Only read and parse config file
  -c, --control=<ctlpath>                                    Set syslog-ng control socket, default=**${localstatedir}**/syslog-ng.ctl
  -i, --interactive                                          Enable interactive mode
  --worker-threads=<max>                                     Set the number of I/O worker threads

My guess is that the persist-file would also contain a variable name ${prefix} if the localstatedir hadn't been set directly in the config.
It is actually more like a question if this is intended or not, although I think it is not.
Debian ticket is: [962839](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=962839)

## Steps to reproduce
Run syslog-ng --help

## Configuration
N/A

### PR DESCRIPTION
**Pull request by bazsi**
This one fixes #4138, which was forwarded by @mochrul from Debian ticket 

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=962839

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

File: syslog-ng/main.c [hunk 1]
```diff
@@ -217,6 +228,7 @@ main(int argc, char *argv[])
   g_process_set_argv_space(argc, (gchar **) argv);
 
   resolved_configurable_paths_init(&resolvedConfigurablePaths);
+  __resolve_paths_in_help_texts();
```
Comment (MrAnno) [inline syslog-ng/main.c:231]: Can't we just move `syslogng_options` to the stack (into a function call that is responsible for the initialization of the command line options), and call `get_installation_path_for()` directly on the `PATH_SYSLOG_NG_CONF`, `SYSLOG_NG_MODULE_PATH`, etc. constants inside the option description?

If you do not prefer that method, can we just remove the double dash prefix of `__resolve_paths_in_help_texts` -> `_resolve_paths_in_help_texts`?

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
@kira-syslogng retest this please

**Review by MrAnno — APPROVED** [review]
Thank you.

**Comment by MrAnno** [comment]
I've followed up the `resolvedConfigurablePaths` rename in our internal modules too.

**Pull request merged**